modules/useQueries.js

```diff
@@ -26,25 +28,40 @@ function useQueries(createHistory) {
       parseQueryString = defaultParseQueryString
 
     function addQuery(location) {
-      if (location.query == null)
-        location.query = parseQueryString(location.search.substring(1))
+      if (location.query == null) {
+        const { search } = location
+        location.query = parseQueryString(search.substring(1))
+        location[SEARCH_BASE_KEY] = { search, searchBase: '' }
```

Comment:
Awesome, thanks for the detailed explanation.

I think ultimately `useQueries` should probably just ignore any existing `search` value and we should alter that test to make sure the existing query string is overwritten. I think this is a lot clearer in the object-based API where everything is in the same argument. I understand you'd like to keep backwards compat for now, but let's def drop in 2.x.